Sxmo 1.6 migraton

ArenM opened the pull request:
This is the work I've done on updating the sxmo packages for 1.6. This should all build, and is mostly working, but still has a couple issues.

The upgrading process is a bit funky. Current sxmo images don't have a locale properly configured which will prevent foot (the terminal used in swmo) from running, so one must be manually installed / configured in the process of upgrading. Also `bemenu` will need to be replaced with `sxmo-bemenu` which pacman will generate a warning about.

This still has a couple of issues, but I don't think this is the correct place to fix them. Suspend works very inconsistently, but the symptoms seem to be pointing at a race condition in the kernel (I'll post some logs in a comment). And the `modemmonitortoggle` script (used for the ModemReset and ModemToggle items in the menu) is partially broken, but will require rewriting most of the script because it makes some bad assumptions about systemctl.

Updated packages:
 - [x] sxmo-utils (mostly working, more patches likely needed)
 - [x] danctnix-sxmo-ui-meta

New packages:
 - [x] sxmo-bemenu (https://git.sr.ht/~stacyharper/bemenu)
 - [x] wvkbd (https://github.com/jjsullivan5196/wvkbd)
 - [x] tinydm (https://gitlab.com/postmarketOS/tinydm)
 - [x] wayout (https://git.sr.ht/~proycon/wayout)
 - [x] foot #240 
 - [x] mmsd-tng #239 
 - [x] sxmo-sway, sxmo-wlroots (https://git.sr.ht/~stacyharper/sway/)

Removed Packages:
 - [x] sxmo-xdm-config

ArenM commented:
The suspend issues I observed seem strange, for now I'm just going to post log, but it might be better to open an issue for further discussion.

These are dmesg logs taken while running rtcwake (what sxmo uses internally to suspend the phone) and sytemctl. The output from rtcwake it's self is usally something like
```
rtcwake: assuming RTC uses UTC ...
rtcwake: wakeup from "freeze" using /dev/rtc0 at Thu Nov 25 02:35:56 2021
rtcwake: write error
```

<details>
<summary>log 1: rtcwake -m mem -s 30</summary>

```
[37464.446544] PM: suspend entry (deep)
[37464.447018] Filesystems sync: 0.000 seconds
[37464.453131] Freezing user space processes ... (elapsed 0.004 seconds) done.
[37464.457957] OOM killer disabled.
[37464.457973] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[37464.459945] printk: Suspending console(s) (use no_console_suspend to debug)
[37464.468432] musb-sunxi 1c19000.usb: Error unknown readb offset 112
[37464.620940] sunxi-mmc 1c0f000.mmc: data error, sending stop command
[37464.621122] sunxi-mmc 1c0f000.mmc: send stop command failed
[37464.621167] mmc0: error -110 reading status reg of PM func
[37464.621245] PM: dpm_run_callback(): mmc_bus_suspend+0x0/0x80 returns -110
[37464.621357] mmcblk mmc0:aaaa: PM: failed to suspend async: error -110
[37464.636962] PM: Some devices failed to suspend, or early wake event detected
[37464.691654] musb-sunxi 1c19000.usb: Error unknown writeb offset 112
[37464.823138] OOM killer enabled.
[37464.823216] Restarting tasks ... done.
[37464.849992] PM: suspend exit
[37465.189973] audit: type=1130 audit(1637789439.569:350): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-rfkill comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[37465.433527] Bluetooth: hci0: RTL: examining hci_ver=07 hci_rev=000b lmp_ver=07 lmp_subver=8703
[37465.438435] Bluetooth: hci0: RTL: chip_type status=0 type=5
[37465.444508] Bluetooth: hci0: RTL: rom_version status=0 version=1
[37465.444758] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723cs_xx_fw.bin
[37465.445231] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723cs_xx_config.bin
[37465.493245] Bluetooth: hci0: RTL: cfg_sz 63, total sz 19427
[37465.871481] Bluetooth: hci0: RTL: fw version 0xaa5ca4dc
[37465.920505] sun8i-ce 1c15000.crypto: Fallback for ecb-aes-sun8i-ce is ecb-aes-ce
```

</details>


<details>
<summary>log 2: systemctl suspend (failed)</summary>

```
[37908.750290] PM: suspend entry (deep)
[37909.049366] Filesystems sync: 0.299 seconds
[37909.050323] Freezing user space processes ... (elapsed 0.002 seconds) done.
[37909.053064] OOM killer disabled.
[37909.053071] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[37909.054563] printk: Suspending console(s) (use no_console_suspend to debug)
[37909.061520] musb-sunxi 1c19000.usb: Error unknown readb offset 112
[37909.214129] sunxi-mmc 1c0f000.mmc: data error, sending stop command
[37909.214290] sunxi-mmc 1c0f000.mmc: send stop command failed
[37909.214328] mmc0: error -110 reading status reg of PM func
[37909.214387] PM: dpm_run_callback(): mmc_bus_suspend+0x0/0x80 returns -110
[37909.214527] mmcblk mmc0:aaaa: PM: failed to suspend async: error -110
[37909.234204] PM: Some devices failed to suspend, or early wake event detected
[37909.289060] musb-sunxi 1c19000.usb: Error unknown writeb offset 112
[37909.417812] OOM killer enabled.
[37909.417914] Restarting tasks ... done.
[37909.472060] PM: suspend exit
[37909.474896] PM: suspend entry (s2idle)
[37909.655355] audit: type=1130 audit(1637789884.072:375): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-rfkill comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[37909.694861] Filesystems sync: 0.219 seconds
[37909.699815] Freezing user space processes ... (elapsed 0.014 seconds) done.
[37909.714234] OOM killer disabled.
[37909.714278] Freezing remaining freezable tasks ... (elapsed 0.004 seconds) done.
[37909.718809] printk: Suspending console(s) (use no_console_suspend to debug)
[37909.895718] musb-sunxi 1c19000.usb: Error unknown readb offset 112
[37910.046304] sunxi-mmc 1c0f000.mmc: data error, sending stop command
[37910.046466] sunxi-mmc 1c0f000.mmc: send stop command failed
[37910.046515] mmc0: error -110 reading status reg of PM func
[37910.046607] PM: dpm_run_callback(): mmc_bus_suspend+0x0/0x80 returns -110
[37910.046708] mmcblk mmc0:aaaa: PM: failed to suspend async: error -110
[37910.064990] PM: Some devices failed to suspend, or early wake event detected
[37910.120974] musb-sunxi 1c19000.usb: Error unknown writeb offset 112
[37910.255164] OOM killer enabled.
[37910.255573] Restarting tasks ... done.
[37910.279752] PM: suspend exit
[37910.439363] audit: type=1130 audit(1637789884.856:376): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-suspend comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
[37910.888303] Bluetooth: hci0: RTL: examining hci_ver=07 hci_rev=000b lmp_ver=07 lmp_subver=8703
[37910.893921] Bluetooth: hci0: RTL: chip_type status=0 type=5
[37910.898592] Bluetooth: hci0: RTL: rom_version status=0 version=1
[37910.898760] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723cs_xx_fw.bin
[37910.899265] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723cs_xx_config.bin
[37910.925743] Bluetooth: hci0: RTL: cfg_sz 63, total sz 19427
[37911.361840] Bluetooth: hci0: RTL: fw version 0xaa5ca4dc
[37911.413162] sun8i-ce 1c15000.crypto: Fallback for ecb-aes-sun8i-ce is ecb-aes-ce
[37912.827490] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes readyc
```
</details>

It's worth nothing here that I didn't change anything between these runs with systemctl, systemctl either changed something, or it worked the second time by random chance.

<details>
<summary>log 2: systemctl suspend (successful)</summary>

```
[37961.079089] PM: suspend entry (deep)
[37961.411255] Filesystems sync: 0.332 seconds
[37961.411919] Freezing user space processes ... (elapsed 0.002 seconds) done.
[37961.414773] OOM killer disabled.
[37961.414781] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[37961.416314] printk: Suspending console(s) (use no_console_suspend to debug)
[37961.423055] musb-sunxi 1c19000.usb: Error unknown readb offset 112
[37961.575022] sunxi-mmc 1c0f000.mmc: data error, sending stop command
[37961.575144] sunxi-mmc 1c0f000.mmc: send stop command failed
[37961.575171] mmc0: error -110 reading status reg of PM func
[37961.575252] PM: dpm_run_callback(): mmc_bus_suspend+0x0/0x80 returns -110
[37961.575373] mmcblk mmc0:aaaa: PM: failed to suspend async: error -110
[37961.591253] PM: Some devices failed to suspend, or early wake event detected
[37961.646013] musb-sunxi 1c19000.usb: Error unknown writeb offset 112
[37961.782105] OOM killer enabled.
[37961.782190] Restarting tasks ... done.
[37961.789896] PM: suspend exit
[37961.791313] PM: suspend entry (s2idle)
[37961.948844] Filesystems sync: 0.157 seconds
[37961.952463] Freezing user space processes ... (elapsed 0.011 seconds) done.
[37961.964703] OOM killer disabled.
[37961.964747] Freezing remaining freezable tasks ... (elapsed 0.004 seconds) done.
[37961.969067] printk: Suspending console(s) (use no_console_suspend to debug)
[37962.278035] musb-sunxi 1c19000.usb: Error unknown readb offset 112
[37978.264908] sunxi-rsb 1f03400.rsb: RSB running at 4000000 Hz
[37978.702539] musb-sunxi 1c19000.usb: Error unknown writeb offset 112
[37978.799300] OOM killer enabled.
[37978.799394] Restarting tasks ... done.
[37978.855375] PM: suspend exit
[37979.008638] audit: type=1130 audit(1637789953.423:385): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-rfkill comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[37979.154923] audit: type=1130 audit(1637789953.567:386): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-suspend comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[37979.162938] audit: type=1131 audit(1637789953.571:387): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-suspend comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[37979.464292] Bluetooth: hci0: RTL: examining hci_ver=07 hci_rev=000b lmp_ver=07 lmp_subver=8703
[37979.471727] Bluetooth: hci0: RTL: chip_type status=0 type=5
[37979.504567] Bluetooth: hci0: RTL: rom_version status=0 version=1
[37979.504821] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723cs_xx_fw.bin
[37979.505959] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723cs_xx_config.bin
[37979.531829] Bluetooth: hci0: RTL: cfg_sz 63, total sz 19427
[37979.917203] Bluetooth: hci0: RTL: fw version 0xaa5ca4dc
[37979.970098] sun8i-ce 1c15000.crypto: Fallback for ecb-aes-sun8i-ce is ecb-aes-ce
[37981.538684] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
```
</details>

Danct12 commented on PKGBUILDS/danctnix/gojq/PKGBUILD:
```diff
@@ -0,0 +1,43 @@
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/itchyny/gojq/archive/v${pkgver}.tar.gz")
+sha256sums=('69616c38cc9bcdef06d32692e1cc1f330980c523fc777b07a851a52b51fffdac')
+
+prepare(){
```
```suggestion
prepare() {
```

Danct12 commented on PKGBUILDS/sxmo/danctnix-sxmo-settings/danctnix-sxmo-settings.install:
```diff
@@ -1,13 +1,45 @@
+  fi
+
+  if [ $(vercmp 1.5.3 1.6) -lt 0 ]; then
+    echo "sxmo no longer depends on some packages, you should remove unused packages"
```
Which package to be exact?

ArenM replied:
xdm is no longer uesd, and if the user decides to use swmo all of the packages that `sxmo-ui-dwm-meta` depends on can be removed as well.

ArenM commented:
I've fixed the formatting issue you found in gojq. I also added a dependency on `xorg-xwayland` to `sxmo-ui-sway-meta`, removed the `sxmo-xdm-config` package, and fixed typos in the commit messages for `sxmo-sway` and `danctnix-sxmo-ui-meta`.

ArenM commented:
Sicne mako seems to be broken on archlinux arm and dunst supports wayland, I decided to use it instead for this swmo build until I can figure out what's wrong with mako.

Danct12 commented:
LGTM
